packages/core/src/routes/interaction/actions/submit-interaction.ts

```diff
@@ -0,0 +1,123 @@
+import type { User } from '@logto/schemas';
+import { Event } from '@logto/schemas';
+import type { Provider } from 'oidc-provider';
+
+import { getLogtoConnectorById } from '#src/connectors/index.js';
+import { assignInteractionResults } from '#src/lib/session.js';
+import { encryptUserPassword, generateUserId, insertUser } from '#src/lib/user.js';
+import { findUserById, updateUserById } from '#src/queries/user.js';
+
+import type {
+  InteractionContext,
+  Identifier,
+  VerifiedInteractionResult,
+  SocialIdentifier,
+  VerifiedSignInInteractionResult,
+  VerifiedRegisterInteractionResult,
+} from '../types/index.js';
+
+const getSocialUpdateProfile = async ({
+  user,
+  connectorId,
+  identifiers,
+}: {
+  user?: User;
+  connectorId: string;
+  identifiers?: Identifier[];
+}) => {
+  // TODO: @simeng refactor me. This step should be verified by the previous profile verification cycle Already.
+  // Should pickup the verified social user info result automatically
+  const socialIdentifier = identifiers?.find(
+    (identifier): identifier is SocialIdentifier =>
+      identifier.key === 'social' && identifier.connectorId === connectorId
+  );
+
+  if (!socialIdentifier) {
+    return;
+  }
+
+  const {
+    metadata: { target },
+    dbEntry: { syncProfile },
+  } = await getLogtoConnectorById(connectorId);
+
+  const { userInfo } = socialIdentifier;
+  const { name, avatar, id } = userInfo;
+
+  const profileUpdate =
+    syncProfile || !user // Always sync profile on register
+      ? {
+          ...(name ? { name } : undefined),
+          ...(avatar ? { avatar } : undefined),
+        }
+      : undefined;
+
+  return {
+    identities: { ...user?.identities, [target]: { userId: id, details: userInfo } },
```

Comment:
Shouldn't we also update the identities from social?

Reply:
Here identities are from the social response.

Reply:
Ah, OK... So it means the identities will update every time whatsoever